src/main/java/mate/academy/springboot/datajpa/repositories/specification/product/ProductCategoryInSpecification.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.springboot.datajpa.repositories.specification.product;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.JoinType;
+import mate.academy.springboot.datajpa.models.Category;
+import mate.academy.springboot.datajpa.models.Product;
+import mate.academy.springboot.datajpa.repositories.specification.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductCategoryInSpecification implements
+        SpecificationProvider<Product> {
+    private static final String FILTER_KEY = "categoryIn";
+    private static final String FIELD_NAME = "name";
+
+    @Override
+    public Specification<Product> getSpecification(String[] categories) {
+        return (root, query, cb) -> {
+            Join<Product, Category> join = root.join("category", JoinType.INNER);
+            CriteriaBuilder.In<String> predicate = cb.in(join.get(FIELD_NAME));
+            for (String category :categories) {
+                predicate.value(category);
+            }
+            return cb.and(predicate, predicate);

```

Comment:
Remove the duplicate 'predicate' in the 'cb.and()' function, as it's not needed.